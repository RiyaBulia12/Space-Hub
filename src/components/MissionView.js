import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions, changeStatus } from '../redux/slices/missionReducer';

function MissionView() {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchMissions);
    // eslint-disable-next-line
  }, []);

  const toggleStatus = (id) => {
    dispatch(changeStatus(id));
  };

  return (
    <div id="missions-section" className="my-5 mx-20 overflow-hidden">
      <table className=" border-collapse table-auto border-2 border-slate-200 ">
        <thead>
          <tr className="">
            <th className="border-2  border-slate-200 p-2">Mission</th>
            <th className="w-4/6 border border-slate-200">Description</th>
            <th className="border-2 border-slate-200 ">Status</th>
            <th className="border-2 border-slate-200 "> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <td className="p-2 border-2 border-slate-200 text">
                {mission.mission_name}
              </td>

              <td className="p-2 border-2 border-slate-200">{mission.description}</td>

              <td className="p-3 border-2 w-[9%] text-center border-slate-200 ">
                {!mission.reserved && <span className="bg-[#6D757D] p-1 px-1.5 text-white rounded-md font-semibold">NOT A MEMBER</span>}
                {mission.reserved && (
                  <span className="bg-[#18A2B8] p-1 px-1.5 text-white rounded-md font-semibold">Active Member</span>
                )}
              </td>

              <td className="p-3 border-2 border-slate-200 text-center w-1/12">
                <button
                  type="button"
                  className="p-2 border-2 border-slate-400 rounded-md"
                  onClick={() => toggleStatus(mission.mission_id)}
                  style={{
                    borderColor: mission.reserved ? '#d90429' : '#343a40',
                    color: mission.reserved ? '#d90429' : '#343a40',
                  }}
                >
                  {mission.reserved ? 'Leave Mission' : 'Join Mission'}
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}

export default MissionView;
