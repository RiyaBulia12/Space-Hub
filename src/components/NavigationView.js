import { NavLink } from 'react-router-dom';
import logo from '../assets/imgs/planet.png';

const NavigationView = () => (
  <>
    <nav className="flex justify-between flex-col mx-4 py-5 md:border-b md:flex-row lg:mx-24">
      <div className="flex items-center px-4">
        <img src={logo} className="h-12 px-4 lg:h-16" alt="Space Hub Logo" />
        <h1 className="text-2xl font-medium lg:text-3xl">Space Traveler&apos;s Hub</h1>
      </div>
      <ul className="self-center">
        <li>
          {[
            ['Rockets', '/rockets'],
            ['Missions', '/missions'],
          ].map(([title, url]) => (
            <NavLink to={url} className="px-8 py-2 text-blue-500 md:text-lg hover:underline" key={title}>{title}</NavLink>
          ))}
        </li>
      </ul>
    </nav>
  </>
);

export default NavigationView;
