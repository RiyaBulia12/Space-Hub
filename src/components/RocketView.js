import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import fetchRockets from '../middleware/api';

const RocketView = () => {
  const rocketsObj = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [rocketsObj.length, dispatch]);

  return (
    <>
      { rocketsObj.map((item) => (
        <div key={item.id} className="mx-8 lg:mx-36">
          <div className="flex flex-col bg-white border shadow-md my-8 md:flex-row text-neutral-700">
            <img className="object-cover w-full h-72 md:h-auto md:w-64" src={item.flickr_images} alt={item.rocket_name} />
            <div className="flex flex-col p-4 leading-normal">
              <h5 className="mb-2 text-2xl font-bold">{item.rocket_name}</h5>
              <p className="mb-3 font-normal ">{item.description}</p>
              <button
                type="button"
                className="w-36 px-4 py-2.5 bg-blue-600 text-white text-sm
              rounded shadow-md hover:bg-blue-700 hover:shadow-lg transition duration-150 ease-in-out"
              >
                Reserve Rocket
              </button>
            </div>
          </div>
        </div>
      ))}
    </>
  );
};

export default RocketView;
