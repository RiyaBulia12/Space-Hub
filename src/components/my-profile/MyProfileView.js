import React from 'react';
import JoinedMissions from './JoinedMissions/JoinedMissions';

const MyProfileView = () => (
  <div className="mx-[8%] my-10">
    <JoinedMissions />
  </div>
);

export default MyProfileView;
