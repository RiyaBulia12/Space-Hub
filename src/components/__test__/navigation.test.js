import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import NavigationView from '../NavigationView';
import MyProfileView from '../my-profile/MyProfileView';
import MissionView from '../MissionView';
import RocketView from '../RocketView';

describe('Navigation', () => {
  it('renders navigation menu correctly', () => {
    const MockNavigation = () => (
      <Provider store={store}>
        <NavigationView />
      </Provider>
    );
    expect(MockNavigation).toMatchSnapshot();
  });

  it('renders rockets correctly', () => {
    const MockNavigation = () => (
      <Provider store={store}>
        <RocketView />
      </Provider>
    );
    expect(MockNavigation).toMatchSnapshot();
  });

  it('renders missions correctly', () => {
    const MockNavigation = () => (
      <Provider store={store}>
        <MissionView />
      </Provider>
    );
    expect(MockNavigation).toMatchSnapshot();
  });

  it('renders my profile correctly', () => {
    const MockNavigation = () => (
      <Provider store={store}>
        <MyProfileView />
      </Provider>
    );
    expect(MockNavigation).toMatchSnapshot();
  });
});
