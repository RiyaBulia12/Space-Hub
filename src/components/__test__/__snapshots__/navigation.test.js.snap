// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders missions correctly 1`] = `[Function]`;

exports[`Navigation renders my profile correctly 1`] = `[Function]`;

exports[`Navigation renders navigation menu correctly 1`] = `[Function]`;

exports[`Navigation renders rockets correctly 1`] = `[Function]`;
